src/util/PathWeight.h

```diff
@@ -0,0 +1,73 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#ifndef UTIL_PATH_WEIGHT_H_
+#define UTIL_PATH_WEIGHT_H_
+
+#include "common/base/Base.h"
+#include "common/datatypes/Value.h"
+#include "common/datatypes/Edge.h"
+
+namespace nebula {
+namespace graph {
+
+class QueryContext;
+class IWeight {
+public:
+    explicit IWeight(QueryContext* qctx) : qctx_(qctx) {}
+    virtual double getWeight(Edge& edge, Value& src, const Value& dst) = 0;
```

Comment:
Why to use different qualifiers for parameter types?  Aren't the src/dst vertex types `Vertex` ?